src/os/bluestore/BlueStore.cc

```diff
@@ -1750,6 +1752,12 @@ int BlueStore::mkfs()
   {
     dout(20) << __func__ << " initializing freespace" << dendl;
     KeyValueDB::Transaction t = db->get_transaction();
+    {
+      bufferlist bl;
+      bl.append(freelist_type);
+      t->set(PREFIX_SUPER, "freelist_type", bl);
+    }
+    fm->create(t);
```

Comment:
Two problems here:
1. it shall be ```fm->create(freelist_type);``` instead of ```fm->create(t);```
2. I think you shall move the initialization of ```fm``` ahead, before we call ``` _open_alloc```, which is intend for space initialization and will access ```fm``` .